core/src/main/kotlin/xtdb/api/Xtdb.kt

```diff
@@ -59,6 +71,26 @@ object Xtdb {
     @JvmOverloads
     fun openNode(config: Config = Config()) = config.open()
 
+    @JvmStatic
+    fun openNode(path: Path): IXtdb {
+        if (path.extension != "yaml") {
+            throw IllegalArgumentException("Invalid config file type - must be '.yaml'")
+        } else if (!path.toFile().exists()) {
+            throw IllegalArgumentException("Provided config file does not exist")
+        }
+
+        val yamlString = Files.readString(path)
+        val config = nodeConfig(yamlString)
+
+        return config.open()
+    }
+
+    @JvmStatic
```

Comment:
I'd drop this, it's easy/well-known enough for Java/Kotlin folks to call `toPath()`